### ISSUE TITLE
How to publish on Azure .NET 8 Blazor Web App with Interactive Auto on CLI?

### ISSUE THREAD

**Issue by MetaHex** [dotnet/aspnetcore#53329]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

This is repost from https://github.com/dotnet/AspNetCore.Docs/issues/31392 . Luke asked me to repost here to determine which team a CLI-Azure publish problem should be reported to or ask them if they'll move your issue to the correct GH repo. They don't work on the CLI or directly on Azure and that error looks like some kind of Azure auth problem. They won't be able to help directly, but they should be able to get you to the right spot.

Issue description: 

Prior to dotnet 8, I can publish from cmd line using:
-dotnet build -c release
-dotnet publish -c release
-on vscode, right click on project folder, 'Deploy of Web app..."

Deployments will fail since dotnet 8. (existing .net7 projects also doesn't work anymore)

output window shows:

Starting deployment...
Creating zip package...
Zip package size: 1.21 GB
Error: Failed to get status of deployment.

I also have issue publishing using visual studio 2022 preview. It works fine on visual studio 2022 though.

Few other information:
-I don't think it's a publish issue in the sense that I can publish on vs2022, but not in cli.
-I don't think it's a .net 8 issue because .net7 projects that used to be able to publish using cli no longer works either.

Looks like the tooling for vs code extension (Azure App Service) hasn't updated since "Last updated | 10/11/2023, 3:02:01 PM". "dotnet publish -c release" should be pretty common command so I am not sure why this isn't working out of the box. I wonder if this is something limited to my environment or it's common. From few online forums, seems a consistent issue. I could try to dig more later, like compare the release between vs2022 vs cli.

The ask is to have a guidelines/documentation to be able to publish using cli.

Fails on both Windows and Mac (Apple Si).

cc: @guardrex https://github.com/dotnet/AspNetCore.Docs/issues/31392

### Expected Behavior

-dotnet build -c release
-dotnet publish -c release
-on vscode, right click on project folder, 'Deploy of Web app..."
-web app deployed successfully on Azure App Service

### Steps To Reproduce

Create a blazor webapp from template: 
-example: dotnet new blazor -o BlazorApp -int Auto

Build and Publish:
-dotnet build -c release
-dotnet publish -c release
-on vscode, right click on project folder, 'Deploy of Web app..."

Result: deployment failed

Output window:

Starting deployment...
Creating zip package...
Zip package size: 1.21 GB (mac/AppleSi) ; on Windows (Intel) this would be (3xxMB)
Error: Failed to get status of deployment.

### .NET Version

8.0.100

**Comment by MetaHex** [comment]
Any guidelines here? Still can't publish on mac

**Comment by MetaHex** [comment]
Since Mac removed VS for Mac, this has become more apparent issue.

Another way to ask this, what's the difference between 'publish' in visual studio vs run dotnet publish -c release command? Is there a way we can compare them?

**Comment by MetaHex** [comment]
For anyone else running into this issue deploying. I think the following happened:

When deploying using vscode, azure extension, the deployment is trying to use deploy with git (local or github or azure devops...). In the past, we could build, publish locally and deploy only the zip from zipdeploy. For some reason, when we publish now (some time after net7), we still do build, publish locally and deploy. But on server, it will still try to compile from git on the server. Meaning, the zipdeploy will need the full git instead of the trimmed. This results that the zip will be very large in size. As a reference, a default app scaffold from "dotnet new blazor -o BlazorApp", the zip generated on mac is 1.2GB and on windows 300+MB. 

tldr; a workaround: using zipdeploy: https://learn.microsoft.com/en-us/azure/app-service/deploy-zip?view=azure-devops&tabs=cli#create-a-project-zip-package

Step 1: add web.confg at root of project if you don't have one:
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <system.webServer>
    <handlers>
      <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModuleV2" resourceType="Unspecified" />
    </handlers>
    <aspNetCore processPath="dotnet" arguments=".\<YOURPROJNAME>.dll" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" />
  </system.webServer>
</configuration>

Step 2: build and publish release
dotnet build -c release
dotnet publish -c release

Step 3: generate zipdeploy
-go to bin\release\.net8\publish and run the following based on Bash or PowerShell

# Bash
zip -r <file-name>.zip .

# PowerShell
Compress-Archive -Path * -DestinationPath <file-name>.zip

Step 4: 
In the browser, navigate to https://<app_name>.scm.azurewebsites.net/ZipDeployUI.

Upload the ZIP package you created in [Create a project ZIP package](https://learn.microsoft.com/en-us/azure/app-service/deploy-zip?view=azure-devops&tabs=cli#create-a-project-zip-package) by dragging it to the file explorer area on the web page.

**Comment by MetaHex** [comment]
If I publish directly from vscode to app service, including create the resources from scratch, the deployment would succeed but unable to boot. I think it's related to this "https://stackoverflow.com/questions/48853599/how-to-debug-you-do-not-have-permission-to-view-this-directory-or-page". Someone posted there:

I deployed straight from VS Code to Azure App Service with no pipelines set up....

If I look at kudo, looks like the unpack was all wrong. Actually, it's just bunch of original source code, not executables. Maybe it expects us to build from scratch on the host?  I feel there's some missing documentation somewhere.

**Comment by javiercn** [comment]
@MetaHex thanks for contacting us.

@danroth27 can you forward this to the right folks. We don't believe there is an ASP.NET specific publish aspect to it.

**Comment by danroth27** [comment]
@MetaHex I think you want to report this issue with the Azure App Service VSCode extension and Azure CLI in the https://github.com/Microsoft/vscode-azureappservice repo. They should be able to direct from there.

@nturinski FYI
